_posts/2021-01-21-ooops-can-I-try-again-please.md

```diff
@@ -0,0 +1,73 @@
+---
+layout: post
+header_image: /img/posts/ooops-can-I-try-again-please/header.jpg
+title: "Ooops, can I try again, please?"
+author: Mauro Servienti
+synopsis: "When systems fail, we can retry the whole process and be successful. However, there are scenarios in which retrying a subset of the process might be a better choice. Not all failures are born equal."
+tags:
+- Distributed Systems
+- Retries
+- Errors
+- Failures
+---
+
+It failed. Systems fail all the time: Bugs, unavailability of required resources, hardware failures, and many more reasons cause systems to fail. When the system uses messages and queues to exchange information and drive behaviors, one of the advantages is that we can retry messages. A message comes in, and the message handling process fails. The message can be returned to the queue and tried again. If message processing logic changes some data, it can rollback the corresponding database transaction. So far, so good, in many scenarios, we can retry messages.
+
+> More information about errors, failed messages, and retries are available in my ["Businesses don't fail, they make mistakes"](https://milestone.topics.it/2019/09/10/businesses-dont-fail-they-make-mistakes.html) and in David Boike's excellent ["I caught an exception. Now what?"](https://particular.net/blog/but-all-my-errors-are-severe) article on Particular's blog.
+
+Is this it? Or is there something more that we can do? There are use cases in which not going through the incoming message retry process is beneficial. Moreover, there are use cases where there is no incoming message to retry, but there is only an outgoing message. Let's start with the latter.
```

Comment:
The use case definitions are a bit abstract. Can you think of simple examples to include?

Reply:
Can you have another look? Thanks.